android/engine/src/main/java/org/smartregister/fhircore/engine/cql/Evaluator.kt

```diff
@@ -0,0 +1,167 @@
+        val libraryStream: InputStream = ByteArrayInputStream(libraryData.toByteArray())
+        val fhirHelpersStream: InputStream = ByteArrayInputStream(helperData.toByteArray())
+        val library = parser.parseResource(libraryStream)
+        val fhirHelpersLibrary = parser.parseResource(fhirHelpersStream)
+        val resources: List<IBaseResource> = Lists.newArrayList(library, fhirHelpersLibrary)
+        val bundleFactory = fhirContext.newBundleFactory()
+        val bundleLinks = BundleLinks("", null, true, BundleTypeEnum.COLLECTION)
+        bundleFactory.addRootPropertiesToBundle(
+            "bundled-directory",
+            bundleLinks,
+            resources.size,
+            null
+        )
+        bundleFactory.addResourcesToBundle(
+            resources, BundleTypeEnum.COLLECTION, "",
+            BundleInclusionRule.BASED_ON_INCLUDES, null
+        )
+        contentProvider = BundleFhirLibraryContentProvider(
+            fhirContext,
+            bundleFactory.resourceBundle as IBaseBundle,
+            adapterFactory,
+            libraryVersionSelector
+        )
+
+        // Load terminology content, and create a TerminologyProvider which is the interface used by the evaluator for resolving terminology
+        val valueSetStream: InputStream = ByteArrayInputStream(valueSetData.toByteArray())
+        val valueSetBundle = parser.parseResource(valueSetStream)
+        terminologyProvider = BundleTerminologyProvider(fhirContext, valueSetBundle as IBaseBundle)
+
+        // Load data content, and create a RetrieveProvider which is the interface used for implementations of CQL retrieves.
+        val dataStream: InputStream = ByteArrayInputStream(testData.toByteArray())
+        val dataBundle = parser.parseResource(dataStream)
+        bundleRetrieveProvider = BundleRetrieveProvider(fhirContext, dataBundle as IBaseBundle)
+        bundleRetrieveProvider!!.terminologyProvider = terminologyProvider
+        bundleRetrieveProvider!!.isExpandValueSets = true
+        cqlEvaluator = CqlEvaluator(
+            object : TranslatingLibraryLoader(
+                ModelManager(), listOf(contentProvider),
+                CqlTranslatorOptions.defaultOptions()
+            ) {
+                // This is a hack needed to circumvent a bug that's currently present in the cql-engine.
+                // By default, the LibraryLoader checks to ensure that the same translator options are used to for all libraries,
+                // And it will re-translate if possible. Since translating CQL is not currently possible
+                // on Android (some changes to the way ModelInfos are loaded is needed) the library loader just needs to load libraries
+                // regardless of whether the options match.
+                override fun translatorOptionsMatch(library: Library): Boolean {
```

Comment:
When we discussed in the PR before this wasn't needed b/c we don't have a case where we are loading options that do not match, is that still true? Can we remove this?

Reply:
Thanks @pld for pointing this out. The translatorOptionsMatch is removed.

Reply:
@pld Unfortunately I have to return this line, otherwise the app will crash. See comments below from author.

// **This is a hack needed to circumvent a bug that's currently present in the cql-engine**.
// By default, the LibraryLoader checks to ensure that the same translator options are used to for all libraries,
// And it will re-translate if possible. Since translating CQL is not currently possible
// on Android (some changes to the way ModelInfos are loaded is needed) the library loader just needs to load libraries
// regardless of whether the options match.